packages/beacon-node/src/chain/archiveStore/metrics/historicalStateMetrics.ts

```diff
@@ -0,0 +1,94 @@
+import {BeaconStateTransitionMetrics} from "@lodestar/state-transition";
+import {getMetrics} from "@lodestar/state-transition";
+import {Gauge, Histogram} from "@lodestar/utils";
+import {RegistryMetricCreator} from "../../../metrics/index.js";
+import {QueueMetrics} from "../../../util/queue/options.js";
+import {RegenErrorType} from "../historicalState/types.js";
+
+export type HistoricalStateRegenMetrics = BeaconStateTransitionMetrics & {
+  regenTime: Histogram;
+  loadStateTime: Histogram;
+  stateTransitionTime: Histogram;
+  stateTransitionBlocks: Histogram;
+  stateSerializationTime: Histogram;
+  regenRequestCount: Gauge;
+  regenSuccessCount: Gauge;
+  regenErrorCount: Gauge<{reason: RegenErrorType}>;
+};
+
+export function createHistoricalStateRegenMetrics(metricsRegister: RegistryMetricCreator): HistoricalStateRegenMetrics {
+  return {
+    ...getMetrics(metricsRegister),
```

Comment:
this is not correct, you are adding the state transition metrics here but these are named differently

eg. for `epochTransitionTime` https://github.com/ChainSafe/lodestar/blob/9f5ab00bab35c8d84ece5b184485aa13f81fab7c/packages/state-transition/src/metrics.ts#L15

it's `lodestar_stfn_epoch_transition_seconds` vs. `lodestar_historical_state_stfn_epoch_transition_seconds` we cannot use the same names here, otherwise the metrics would combine which we don't want

Reply:
Sorry I missed the prefix for the `name`, as I was actually comparing the keys e.g. `epochTransitionTime` and `epochTransitionCommitTime`

Reply:
we should dedupe them though, that's the idea of the other refactoring we had in mind https://github.com/ChainSafe/lodestar/issues/7747